server/etcdserver/api/v3rpc/qmon.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 1016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3rpc
+
+import (
+	"context"
+	"fmt"
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	adt "go.etcd.io/etcd/pkg/v3/adt"
+	"go.etcd.io/etcd/server/v3/etcdserver"
+	"golang.org/x/time/rate"
+	"io/ioutil"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	DefaultTotalMemoryBudget            = 1 * 1024 * 1024 * 1024
+	DefaultDegradedBandwidthBytesPerSec = 64 * 1024 * 1024
+	DefaultBurstBytes                   = 64 * 1024 * 1024
```

Comment:
BurstBytes needs to be higher than `DefaultDegradedBandwidthBytesPerSec`. Otherwise, `waitN` will fail.

Reply:
We call NewLimiter as following so burst bytes is bigger than the refill 
rate.NewLimiter(rate.Every(time.Second/time.Duration(qm.degradedBandwidthBytesPerSec)), int(qm.burstBytes))